firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalDocumentsView.java

```diff
@@ -281,4 +320,11 @@ void recalculateAndSaveOverlays(Set<DocumentKey> documentKeys) {
 
     return results;
   }
+
+  /** Returns a base document that can be used to apply `overlay`. */
+  private MutableDocument createBaseDocument(DocumentKey key, @Nullable Overlay overlay) {
```

Comment:
consider renaming to `getBaseDocument()` since we're getting the base doc from remote doc cache, and then creating an invalid one if it's not found

Reply:
Done (this was also the original name)